codec-http/src/main/java/io/netty/handler/codec/http/websocketx/CloseWebSocketFrame.java

```diff
@@ -15,129 +15,134 @@
  */
 package io.netty.handler.codec.http.websocketx;
 
-import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
+import io.netty.buffer.api.Buffer;
+import io.netty.buffer.api.BufferAllocator;
 import io.netty.util.CharsetUtil;
 import io.netty.util.internal.StringUtil;
 
+import java.nio.charset.StandardCharsets;
+
 /**
  * Web Socket Frame for closing the connection.
  */
 public class CloseWebSocketFrame extends WebSocketFrame {
 
     /**
      * Creates a new empty close frame.
+     *
+     * @param allocator {@link BufferAllocator} to use for allocating data.
      */
-    public CloseWebSocketFrame() {
-        super(Unpooled.buffer(0));
+    public CloseWebSocketFrame(BufferAllocator allocator) {
```

Comment:
same comment as above

Reply:
This will still need to allocate an empty buffer somehow. I've created #12021 to simplify that.

Reply:
I was more saying that the user should explicit pass in the buffer always. But we can change that also later